stock_picking_mgmt_weight/wizards/move_weight.py

```diff
@@ -0,0 +1,137 @@
+    move_line_ids = fields.One2many(
+        comodel_name='stock.move.line',
+        inverse_name='move_id')
+    move_line_weight_ids = fields.One2many(
+        comodel_name='stock.move.line.weight.wizard',
+        inverse_name='move_weight_id')
+
+    move_id = fields.Many2one('stock.move')
+
+    weight_selection = fields.Selection([
+        ('theoretical_qty', _('Theoretical quantity demanded')),
+        ('net_weight', _('Origin/Destination Weight'))],
+        default='net_weight')
+
+    theoretical_qty = fields.Float(
+        string='Theoretical quantity demanded',
+        digits='Product Unit of Measure')
+    net_weight = fields.Float(
+        string='Net',
+        digits='Product Unit of Measure')
+
+    weight_selected = fields.Float(
+        digits='Product Unit of Measure',
+        compute='_compute_weight_selected')
+    weight_classified = fields.Float(
+        digits='Product Unit of Measure',
+        compute='_compute_weight_classified',
+        readonly=True,
+        store=True)
+
+    @api.onchange('weight_selection')
+    @api.depends('weight_selected')
+    def _compute_weight_selected(self):
+        if self.weight_selection == 'theoretical_qty':
+            self.weight_selected = self.theoretical_qty
+        else:
+            self.weight_selected = self.net_weight
+
+    @api.depends('move_line_weight_ids.product_uom_qty', 'weight_selected')
+    def _compute_weight_classified(self):
+        self.weight_classified = self.weight_selected - sum(self.move_line_weight_ids.mapped('product_uom_qty'))
+
+    def write_operations(self):
+        self.ensure_one()
+        purchase_order = self.env['purchase.order']
+        if self.weight_classified == 0:
+            if not self.move_line_weight_ids.mapped('order_line'):
+
+                order_new = purchase_order.new({
+                    'partner_id': self.picking_id.partner_id.id
+                })
+                order_new.onchange_partner_id()
+                purchase_order |= purchase_order.create(order_new._convert_to_write(order_new._cache))
+
+                for move_line in self.move_line_weight_ids:
+                    purchase_order.order_line.create({
+                        'name': move_line.product_id.name,
+                        'product_id': move_line.product_id.id,
+                        'product_qty': move_line.product_uom_qty,
+                        'product_uom': move_line.product_uom_id.id,
+                        'price_unit': move_line.product_id.product_tmpl_id.standard_price,
+                        'order_id': purchase_order.id,
+                        'date_planned': datetime.now(),
+                    })
+
+                purchase_order.button_confirm()
+                picking_new = purchase_order.picking_ids
+                picking_new.vehicle_id = self.picking_id.vehicle_id.id
```

Comment:
Better comment this line, as for classification the vehicle is not important, and pickings will be linked